### PR TITLE
Fix ShowBgpInstanceNeighborsDetail to match 'no-prepend' 'replace-as' and 'dual-as'

### PR DESCRIPTION
**Pull request by ancker010**
## Description
The ShowBgpInstanceNeighborsDetail p3 parser wasn't matching for the 'no-prepend' 'replace-as' and 'dual-as' that may be present on the Remote AS/Local AS line. Further, I've added lines to make sure state, either True or False for each flag is added to the output dictionary.

## Motivation and Context
When any of these flags are configured, the parser is unable to gather the Remote AS, Local AS, or any of the flags. And thus doesn't include them in the output dictionary.

## Impact (If any)
N/A

## Screenshots:
Screen Shot 2021-07-29 at 3 24 15 PM
More:
Screen Shot 2021-07-29 at 3 40 20 PM

## Checklist:

- [x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [ ] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [ ] All new code passed compilation.

### CODE REVIEW

**Comment by ancker010** [comment]
I looked through why the Run Test failed....it seems to be related to some iosxe thing that has nothing to do with my commits.
As you can see, the iosxr unittests passed.

Screen Shot 2021-07-29 at 3 57 35 PM

File: src/genie/libs/parser/iosxr/show_bgp.py [hunk 1]
```diff
@@ -2673,10 +2671,18 @@ def cli(self, vrf_type='all', vrf='all', instance='all', neighbor='', address_fa
                     sub_dict['link_state'] = m.groupdict()['link_state']
                 sub_dict['local_as_as_no'] = int(m.groupdict()['local_as_as_no'])
 
-                # Default the values - overwritten if configured
-                sub_dict['local_as_no_prepend'] = False
-                sub_dict['local_as_replace_as'] = False
-                sub_dict['local_as_dual_as'] = False
+                if m.groupdict()['local_as_no_prepend']:
```
Comment (tahigash) [inline src/genie/libs/parser/iosxr/show_bgp.py:2674]: thanks for fixing/updating this. Can I ask you to add UT with output from device?

Comment (ancker010) [reply src/genie/libs/parser/iosxr/show_bgp.py:2675]: Am I mistaken that there's currently no UT or folder based test at all for this command?

Comment (tahigash) [reply src/genie/libs/parser/iosxr/show_bgp.py:2675]: there is as old style UT.
https://raw.githubusercontent.com/CiscoTestAutomation/genieparser/master/src/genie/libs/parser/iosxr/tests/test_show_bgp.py

But if you can, please add UT for this as folder based test.

Comment (ancker010) [reply src/genie/libs/parser/iosxr/show_bgp.py:2675]: I've updated the old-style test, but I really don't have the capacity to make a whole new folder based test for this command.

Comment (ancker010) [reply src/genie/libs/parser/iosxr/show_bgp.py:2675]: Update. I think I've got this done. The folder-based test works on my local system.

Screen Shot 2021-08-03 at 11 03 04 AM

Comment (tahigash) [reply src/genie/libs/parser/iosxr/show_bgp.py:2675]: thanks. yes, only what you developed is fine.

**Review by tahigash — APPROVED** [review]
thanks for addressing comments and contribution.

**Review by lsheikal — APPROVED** [review]

**Comment by lsheikal** [comment]
Thank you for your contribution!!!

**Pull request merged**